mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
I've determined the performance bug is not with the processing of all events, but rather specifically somewhere inside _synti->play(event , syntiIdx), because when I comment that line out, it goes away.  Of course there is a lot caching required by fluidsynth to render the sounds from the wavetables.  But the thing is that even when I remove the soundfont (and thus disable synthesis of the wavetables), I still hear the lag.  So that means somewhere inside od fluidsynth play but not the actual wavetable synthesis is causing the problem.